source/_integrations/input_select.markdown

```diff
@@ -213,3 +213,53 @@ input_select:
 ```
 
 {% endraw %}
+
+Example of `input_select` being used to display the name associated to a numeric state.
+For example, the wallbox *DomBusEVSE* outputs the entity `sensor.evstate` with the following numeric values:
+
+1 -> Car disconnected  
+2 -> Car connected  
+3 -> Charging   
+4 -> Charging + Ventilation req.  
+....  
+
+The following example shows how to create an `input_selector` helper and an automation that updates it when `sensor.evstate` changes: the automation create a list/array *slist* within all options of the input_selector, the variable *item* corresponding to the evse state, and set the input_selector option to *slist[item]*.
+
+{% raw %}
+```yaml
+# define the input_select helper that shows the status name associated to sensor.evstate numeric status
+# the following lines go into configuration.yaml file
+input_select:
+  evstatename:
+    name: EVStateName
+    options:
+    - Off
+    - Disconnected
+    - Connected
+    - Charging
+    - Charging+Vent
+    - Alarm EV
+    - Alarm Power Outage
+    - Alarm Welded
+    initial: Disconnected
+
+# define an automation that update evstatename when sensor.evstate changes
+# the following lines go into automations.yaml file
+# in action, it creates a list variable within all options of the input_select.evstatename
+# and a variable corresponding to the sensor.evstate, then set the proper option for input_select.evstatename
+# it's really simple and works perfect.
+- id: "evstatenameautomation"
+  alias: "Automation that updates EVStateName entity/helper when EVState changes"
```

Comment:
Why would we ever use a automation to update the state of a helper as presented here? This seems to mix up read only vs control?

Reply:
Hello @frenck , 
the purpose of the example is to get a read-only helper that shows the status name corresponding to a read-only status sensor.
In my case, I had a device that returns a status number, and I would have an helper showing me the corresponding status name.
If you think that nobody else will need a similar function, you can refuse the PR.
I believe that HA documentation should be improved: I spent a long time before finding similar solution in a forum post.

Reply:
> the purpose of the example is to get a read-only helper that shows the status name corresponding to a read-only status sensor.

But there is no such thing as a read-only helper; thus this example is invalid.

The correct construct to use in this case, is a template sensor; not an helper and automation combination.